invenio_rdm_records/services/components/internal_notes.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM service component for custom fields."""
+import uuid
+from copy import copy, deepcopy
+from datetime import datetime, timezone
+from invenio_drafts_resources.services.records.components import ServiceComponent
+
+
+class InternalNotesComponent(ServiceComponent):
+    """Service component for custom fields."""
+
+    field = "internal_notes"
+
+    def create(self, identity, data=None, record=None, **kwargs):
+        """Inject note to the record."""
+        notes = data.get(self.field, [])
+        for note in notes:
+            note.update(
+                {
+                    "added_by": {"user": identity.id},
+                    "timestamp": datetime.now(timezone.utc).isoformat(),
+                    "id": str(uuid.uuid4()),
```

Comment:
do we need such a large UUID here? 16 chars?
wouldn't it a 4 chars id enough, or similar?

```python
def _get_node_id(...):
    """Generate a random short id."""
    # import secrets
    return secrets.token_hex(3)
```

Reply:
lets discuss this together with Zach, he suggested the uuid for consistency. I used initially the cool pid.